pkg/usrlocalsharelima/usrlocalsharelima.go

```diff
@@ -27,19 +27,43 @@ import (
 // of the symlink, and not the actual executable. This is important when using Homebrew.
 //
 // If os.Args[0] is invalid, this function still falls back on os.Executable().
-var executableViaArgs0 = sync.OnceValues(func() (string, error) {
+var executableViaArgs0 = sync.OnceValues(func() (self string, _ error) {
+	self, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+
 	if os.Args[0] == "" {
 		logrus.Warn("os.Args[0] has not been set")
-	} else {
-		executable, err := exec.LookPath(os.Args[0])
-		if err == nil {
-			// LookPath() will add the `.exe` file extension on Windows, but will not return an
-			// absolute path if the argument contained any of `:/\` (or just `/` on Unix).
-			return filepath.Abs(executable)
-		}
+		return
+	}
+
+	inPath, err := exec.LookPath(os.Args[0])
+	if err != nil {
 		logrus.Warnf("os.Args[0] is invalid: %v", err)
+		return
 	}
-	return os.Executable()
+
+	// Make sure the path actually matches the running binary.
+	statSelf, err := os.Stat(self)
+	if err != nil {
+		logrus.WithError(err).Warnf("failed to stat running binary: %q", self)
+		return
```

Comment:
Should `os.Stat()` error be real errors? It feels wrong to just ignore them.

But if you want to ignore them, how do you decide if you want to return `self` or `inPath`?

I think it should be an error return:

```suggestion
		return "", fmt.Errorf("failed to stat running binary %q: %w", self, err)
```

Same for the next stat call.